cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/language_specific_display.spec.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+  INDEX_WITH_TIME_2,
+  QueryLanguages,
+  SECONDARY_ENGINE,
+} from '../../../../../utils/constants';
+import {
+  generateAllTestConfigurations,
+  getRandomizedWorkspaceName,
+  getRandomizedDatasourceName,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import {
+  generateDisplayTestConfiguration,
+  getLanguageReferenceTestText,
+} from '../../../../../utils/apps/query_enhancements/language_specific_display';
+
+const workspaceName = getRandomizedWorkspaceName();
+const datasourceName = getRandomizedDatasourceName();
+
+export const runDisplayTests = () => {
+  describe('Language-Specific Display', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.setupTestData(
+        SECONDARY_ENGINE.url,
+        [
+          `cypress/fixtures/query_enhancements/data-logs-1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data-logs-2/${INDEX_WITH_TIME_2}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data-logs-1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data-logs-2/${INDEX_WITH_TIME_2}.data.ndjson`,
+        ]
+      );
+      // Add data source
+      cy.addDataSource({
+        name: datasourceName,
+        url: SECONDARY_ENGINE.url,
+        authType: 'no_auth',
+      });
+
+      // Create workspace
+      cy.deleteWorkspaceByName(workspaceName);
```

Comment:
Small heads up that `cy.deleteWorkspaceByName` and `cy.createInitialWorkspaceWithDatasource` are being moved under the new `cy.osd` namespace in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9150.

If my PR gets merged first I believe there's a one click rebase, otherwise I'll make sure to include the change in my PR.